Doc/reference/datamodel.rst

```diff
@@ -1945,7 +1945,8 @@ Notes on using *__slots__*
   program undefined.  In the future, a check may be added to prevent this.
 
 * Nonempty *__slots__* does not work for classes derived from "variable-length"
-  built-in types such as :class:`int`, :class:`bytes` and :class:`tuple`.
+  built-in types such as :class:`int`, :class:`bytes` and :class:`tuple`
+  (see :c:member:`~PyTypeObject.tp_itemsize`).
```

Comment:
How about something like this? This means that:
- we also clear up the confusion you mentioned in the issue about what "does not work" means
- the link to `tp_itemsize` draws less attention/is more concise

```rst
* :exc:`TypeError` will be raised if nonempty *__slots__* are defined for a
class derived from a
:c:member:`"variable-length" built-in type <PyTypeObject.tp_itemsize>` such as
:class:`int`, :class:`bytes` or :class:`tuple`.

Reply:
This is indeed much better, thanks for the suggestion